op-challenger/fault/types/types.go

```diff
@@ -34,6 +34,20 @@ func (s GameStatus) String() string {
 	}
 }
 
+// GameStatusFromUint8 returns a game status from the uint8 representation.
+func GameStatusFromUint8(i uint8) (GameStatus, error) {
+	switch i {
+	case 0:
+		return GameStatusInProgress, nil
+	case 1:
+		return GameStatusChallengerWon, nil
+	case 2:
+		return GameStatusDefenderWon, nil
+	default:
+		return GameStatus(0), errors.New("invalid game status")
+	}
+}
```

Comment:
I think this is where a cast is fine since the values used for the constants should match up with what the contract uses. Having a dedicated method to do the conversion is probably a good idea, but it can just do the cast - it still provides a single point to handle any difference between the contract values and go ones if needed in the future.